src/items/generics.md

```diff
@@ -0,0 +1,107 @@
+# Type and Lifetime Parameters
```

Comment:
Is there a reason this is named `items/generics.md` and not `types_and_lifetime_parameters.md`?

Reply:
Name length. Maybe to avoid issues once constant parameters are stabilized.